### PR TITLE
Vault cluster name and ID

### PR DESCRIPTION
**Pull request by vishalnayak**
Unfinished business. Do Not Merge.

### CODE REVIEW

File: http/sys_health.go [hunk 1]
```diff
@@ -114,13 +114,28 @@ func getSysHealth(core *vault.Core, r *http.Request) (int, *HealthResponse, erro
 		code = standbyCode
 	}
 
+	// Fetch the local cluster name and identifier
+	var clusterName, clusterID string
+	cluster, err := core.Cluster()
+
+	// Don't set the cluster details in the health status when Vault is sealed
+	if err != nil && err.Error() != "Vault is sealed" {
```
Comment (jefferai) [inline http/sys_health.go]: I think a better approach would be to use the `sealed` boolean from earlier, and simply skip the cluster checks if it's true. That way if we know that the core isn't sealed but get an error anyways, we can just assume it's an internal error (and not rely on string matching).

File: vault/cluster.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	}
+
+	// If cluster name is not supplied, generate one
+	if c.clusterName == "" {
+		clusterNameBytes, err := uuid.GenerateRandomBytes(4)
+		if err != nil {
+			c.logger.Printf("[ERR] core: failed to generate cluster name: %v", err)
+			return err
+		}
+		c.clusterName = fmt.Sprintf("vaule-cluster-%08x", clusterNameBytes)
```
Comment (jefferai) [inline vault/cluster.go]: vaule?  :-D

File: http/sys_health.go [hunk 3]
```diff
@@ -131,4 +146,6 @@ type HealthResponse struct {
 	Standby       bool   `json:"standby"`
 	ServerTimeUTC int64  `json:"server_time_utc"`
 	Version       string `json:"version"`
+	ClusterName   string `json:"cluster_name"`
```
Comment (jefferai) [inline http/sys_health.go]: Let's `omitempty` for these two values unless we hear a complaint about why we shouldn't. We can always add, but it's harder to subtract later.

**Comment by jefferai** [comment]
LGTM after addressing the few comments.

File: vault/cluster.go (hunk 2 above)
Comment (vishalnayak) [inline vault/cluster.go]: Its 'vault' :)

File: http/sys_health.go (hunk 3 above)
Comment (vishalnayak) [inline http/sys_health.go]: Done.

**Pull request merged**